bazel/cc_toolchains/clang_configuration.bzl

```diff
@@ -30,7 +30,10 @@ def _detect_system_clang(repository_ctx):
         version_output = _run(repository_ctx, [cc, "--version"]).stdout
         if not "clang" in version_output:
             fail("The `CC` environment variable is not a Clang compiler.")
-        return repository_ctx.path(cc)
+        clang_path_guess = repository_ctx.path(cc)
+        if clang_path_guess.exists:
+            return(clang_path_guess)
+        return repository_ctx.which(cc)
```

Comment:
Looking at this, here's a variation on the fix, running `--version` on the resolved path:

```
        cc_path = repository_ctx.path(cc)
        if not cc_path.exists:
            cc_path = repository_ctx.which(cc)
        if cc_path and cc_path.exists:
            version_output = _run(repository_ctx, [cc_path, "--version"]).stdout
            if "clang" not in version_output:
                fail("The `CC` environment variable `%s` is not a Clang compiler." % cc)
            return cc_path
```

(slipping in a slight improvement to the error message)

Reply:
Happy to use this if you think it's better than the updated PR, but I think I added a bit more helpful error messages (and allowed failed CC lookup to default to system clang)